tests/baselines/reference/bestCommonTypeOfTuple2.types

```diff
@@ -29,7 +29,7 @@ class F extends C { f }
 class C1 implements base1 { i = "foo"; c }
 >C1 : C1
 >base1 : base1
->i : string
+>i : any

```

Comment:
Yeah something like that is what I meant. In @DanielRosenwasser, a and b should both be number. So for null and undefined, the contextual type should only take effect if you are not in an inference context.